widgets/views/shareLink.php

```diff
@@ -1,32 +1,71 @@
 <?php
 
 use humhub\modules\socialshare\assets\Assets;
-use yii\helpers\Html;
+use humhub\modules\socialshare\models\ConfigureForm;
+use humhub\modules\socialshare\services\SocialShareService;
 
 Assets::register($this);
+
+$settings = new ConfigureForm();
+$settings->loadSettings();
+
+$linkOptions = [
+    'data-ui-widget' => 'socialshare.ShareLink',
```

Comment:
no need to add the attribute `'data-ui-widget' => 'socialshare.ShareLink'` to each link, it is better to add it one time to the wrapper `<span class="shareLinkContainer">`:

- for the class `class ShareLink` add the method:
```php
protected function getAttributes()
{
    return ['class' => 'shareLinkContainer'];
}
```
- then add `'options' => $this->getOptions(),` for `$this->render('shareLink', [...`
- replace `<span class="shareLinkContainer">` with `<?= Html::beginTag('span', $options) ?>`, and `</span>` with `<?= Html::endTag('span') ?>`

Reply:
Done!